RecentActivity/src/org/sleuthkit/autopsy/recentactivity/ChromeCacheExtractor.java

```diff
@@ -0,0 +1,1156 @@
+            this.byteBuffer = buffer;
+        }
+        
+        public RandomAccessFile getFileCopy() {
+            return fileCopy;
+        }
+        public ByteBuffer getByteBuffer() {
+            return byteBuffer;
+        }
+        AbstractFile getAbstractFile() {
+            return abstractFile;
+        } 
+    }
+
+    ChromeCacheExtractor(Content dataSource, IngestJobContext context ) {
+        moduleName = NbBundle.getMessage(ChromeCacheExtractor.class, "ChromeCacheExtractor.moduleName");           
+        this.dataSource = dataSource;
+        this.context = context;
+    }
+    
+    
+    /**
+     * Initializes Chrome cache extractor module.
+     * 
+     * @throws IngestModuleException 
+     */
+    void init() throws IngestModuleException {
+        
+        try {
+            currentCase = Case.getCurrentCaseThrows();
+            fileManager = currentCase.getServices().getFileManager();
+             
+            // Create an output folder to save any derived files
+            absOutputFolderName = RAImageIngestModule.getRAOutputPath(currentCase, moduleName);
+            relOutputFolderName = RAImageIngestModule.getRelModuleOutputPath() + File.separator + moduleName;
+            File dir = new File(absOutputFolderName);
+            if (dir.exists() == false) {
+                dir.mkdirs();
+            }
+        } catch (NoCurrentCaseException ex) {
+            String msg = "Failed to get current case.";
+            throw new IngestModuleException(msg, ex);
+        } 
+    }
+    
+    /**
+     * Cleans up after the module is done.
+     * 
+     * Removes any temp copies of cache files created during extraction.
+     * 
+     */
+    void cleanup () {
+        
+        for (Entry<String, CacheFileCopy> entry : this.filesTable.entrySet()) {
+            String tempFilePathname = RAImageIngestModule.getRATempPath(currentCase, moduleName) + File.separator + entry.getKey(); 
+            try {
+                entry.getValue().getFileCopy().getChannel().close();
+                entry.getValue().getFileCopy().close();
+               
+                File tmpFile = new File(tempFilePathname);
+                if (!tmpFile.delete()) {
+                    tmpFile.deleteOnExit();
+                }
+            } catch (IOException ex) {
+                logger.log(Level.SEVERE, String.format("Failed to delete cache file copy %s", tempFilePathname), ex);
+            }
+        }
+    }
+    
+    /**
+     * Returns the location of output folder for this module
+     * 
+     * @return 
+     */
+    private String getAbsOutputFolderName() {
+        return absOutputFolderName;
+    }
+    
+     /**
+     * Returns the relative location of output folder for this module
+     * 
+     * @return 
+     */
+    private String getRelOutputFolderName() {
+        return relOutputFolderName;
+    }
+    
+    /**
+     * Extracts the data from Chrome cache
+     */
+    void getCache() {
```

Comment:
would it make sense to break getCache into multiple functions?  Maybe put the code in the nested loop into its own function?

Reply:
Yes, this method could and should be simplified and broken  up a little.

I realize that this method is going to need some refactoring in order to handle multiple user profiles ( and hence caches) in a data source.  

Therefore Id like to defer this and address as it as part of that.

Reply:
OK.

Reply:
@raman-bt and @kellykelly3, what is the status of this PR? I see that there are still Codacy issues that can be easily remedied, is this otherwise ready to go?